torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -755,6 +752,84 @@ def test_profiler_with_remote_builtin(self):
             use_record_function=True
         )
 
+    @dist_init
+    def test_async_record_function_double_end_callbacks(self):
+        # We need to disable the internal profiling implementation so we can check
+        # it here without any duplicated calls.
+        _disable_profiling_for_testing()
```

Comment:
@pritamdamania87 , if we removed this, then in the calls to `rpc_async` in `api.py`, we will do what we do in this test: create a `record_function` using the ctx manager and call `rf._call_end_callbacks_on_fut`. The issue is that since we want to test these calls here, we can't call them twice, since that will introduce correctness issues - since we want double calls to `rf._call_end_callbacks_on_fut` to return an error, for example.

Reply:
Hmm, this indeed works with the JIT future, but I didn't quite follow why this is not needed in the case of a JIT future.

The reason there's `_disable_profiling_with_testing()` - consider the test `test_async_record_function_end_callbacks` a bit further down:

If we didn't have the flag, then inside `rpc.rpc_async`, we would have already invoked `rf._call_end_callbacks_on_jit_fut`. However, since that test wants to explicitly test this function, we call it again, but double calls aren't supported. Hence the flag to disable the call in `rpc.rpc_async`.